tests/nested/manual/core20-cloud-init-maas-signed-seed-data/task.yaml

```
@@ -139,53 +167,103 @@ execute: |
   . "$TESTSLIB/nested.sh"
 
   echo "Waiting for snapd to react to cloud-init"
-  seen=0
-  # the first message is about snapd noticing that cloud-init is in error, and
-  # for how long snapd will wait for cloud-init, this message should show up
-  # relatively quickly
-  for i in $(seq 60); do
-      if tests.nested exec "sudo journalctl --no-pager -u snapd" | MATCH "cloud-init reported to be in error state, will disable in 3 minutes"; then
-          seen=1
-          break
-      fi
-      sleep 1
-  done
-  if [ "$seen" = "0" ]; then
-      echo "snapd did not notice cloud-init in error state, test failed"
-      exit 1
-  fi
-  seen=0
-
-  # we now wait for 3 minutes before snapd gives up waiting and disables 
-  # cloud-init
-  for i in $(seq 200); do
-      if tests.nested exec "sudo journalctl --no-pager -u snapd" | MATCH "reported to be in error state after 3 minutes"; then
-          seen=1
-          break
-      fi
-      sleep 1
-  done
-
-  if [ "$seen" = "0" ]; then
-      echo "snapd did not disable cloud-init, test failed"
-      exit 1
+
+  # ugh bash quoting is terrrrrrrrrrible - we can't deeply nest quotes in retry,
+  # tests.nested and MATCH here so we have to copy the command to a script to be
+  # executed instead
+  cat > /tmp/snapd-journal-match-script.sh <<EOF
```

Hmm I think there's something wrong here, you probably want:
```suggestion
  cat > /tmp/snapd-journal-match-script.sh <<'EOF'
```
Otherwise `"$*"` gets evaluated right now, and most likely expands to `""` which will make MATCH always be true?

---

Yes you are correct of course; applied